pkg/util/push/push.go

```diff
@@ -6,19 +6,28 @@ import (
 
 	"github.com/go-kit/kit/log/level"
 	"github.com/weaveworks/common/httpgrpc"
+	"github.com/weaveworks/common/middleware"
 
 	"github.com/cortexproject/cortex/pkg/distributor"
 	"github.com/cortexproject/cortex/pkg/ingester/client"
 	"github.com/cortexproject/cortex/pkg/util"
 )
 
 // Handler is a http.Handler which accepts WriteRequests.
-func Handler(cfg distributor.Config, push func(context.Context, *client.WriteRequest) (*client.WriteResponse, error)) http.Handler {
+func Handler(cfg distributor.Config, sourceIPs *middleware.SourceIPExtractor, push func(context.Context, *client.WriteRequest) (*client.WriteResponse, error)) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		logger := util.WithContext(ctx, util.Logger)
+		if sourceIPs != nil {
```

Comment:
Thinking loudly. Not a big fan of having it here. The context injection could be probably easily done in a generic middleware with which you wrap `push.Handler()`, but I don't have a good solution for the `logger`.

Reply:
Yes, the logger stumped me as well there. I'll have a think about that.

Reply:
We can keep it as is for now.